### PR TITLE
Mod/warmfix 4116

### PR DESCRIPTION
**Pull request by BrianZito**
**Description:**
Changes logic to allow for more complicated subqueries, and allows parsing SQL strings in a more safe and understandable way

**Technical details:**
All parties agree that this is scotch tape, and should be followed up by a more elegant refactor. Ticket for that refactor is being constructed here: https://federal-spending-transparency.atlassian.net/browse/DEV-4184

**Requirements for PR merge:**

1. [ ] Unit & integration tests updated
2. [ ] API documentation updated
3. [ ] Necessary PR reviewers:
    - [ ] Backend
4. [ ] Matview impact assessment completed
5. [ ] Frontend impact assessment completed
6. [ ] Data validation completed
7. [ ] Appropriate Operations ticket(s) created
8. [ ] Jira Ticket [DEV-123](https://federal-spending-transparency.atlassian.net/browse/DEV-123):
    - [ ] Link to this Pull-Request
    - [ ] Performance evaluation of affected (API | Script | Download)
    - [ ] Before / After data comparison

**Area for explaining above N/A when needed:**
```
```

### CODE REVIEW

**Comment by BrianZito** [comment]
Unless you guys really object, I'm going to ignore the codeclimate warning for now. I'm writing a SQL parser, keeping cognitive complexity down is not a realistic goal unless you really want to obfuscate your code.

File: usaspending_api/download/tests/unit/test_file_generation.py [hunk 1]
```diff
@@ -127,119 +127,3 @@ def test_idv_treasury_account_funding_csv_sources():
     VALUE_MAPPINGS["idv_federal_account_funding"]["filter_function"] = original
     assert csv_sources[0].file_type == "treasury_account"
     assert csv_sources[0].source_type == "idv_federal_account_funding"
-
-
-def test_apply_annotations_to_sql_just_values():
```
Comment (tony-sappe) [inline usaspending_api/download/tests/unit/test_file_generation.py]: I'm worried that these tests were removed. Shouldn't they still pass?

Comment (BrianZito) [reply usaspending_api/download/tests/unit/test_file_generation.py]: The sample SQL was not of the format that Django spits out, which the parser relies on. A new test with more representative SQL was added to the new test class, in order to have all annotation tests in the same place. These tests have nothing to do with generating a file.

Comment (tony-sappe) [reply usaspending_api/download/tests/unit/test_file_generation.py]: If it's that fragile and tied to Django, are you concerned with the near-future upgrade to Django 2.X?
https://federal-spending-transparency.atlassian.net/browse/DEV-3074

**Pull request closed without merging**